Add VQVAE_oasis3

Azrail09Code opened the pull request:
This is the submission of the VQVAE generative algorithm by 44777986 (Robert Eldridge). The VQVAE component functions as intended, whereas the generative design is less co-operative. Work could be done to replace the current inadequate PixelCNN generator with an attention-based transformer, and this would likely speed up the process greatly.

The major functions and model declarations are stored in main_functions.py, and the saved models can be tested via running the generate_image.ipynb.

Azrail09Code committed:
Added OASIS image loading functionality

Azrail09Code committed:
Added basic VQVAE without skip connections

Azrail09Code committed:
Updated VQVAE with two-stage vector compression in line with Razavi et al. - SSIM above 0.8

Azrail09Code committed:
Add template README file to the VQVAE folder

Azrail09Code committed:
Add diagram of the two-stage VQVAE algorithm

Azrail09Code committed:
Started write-up and testing Git Markdown Syntax

Azrail09Code committed:
Adding additional images for README documentation

Azrail09Code committed:
Included more information + photos to the README

Azrail09Code committed:
First iteration of Generator design - VQVAE works

Azrail09Code committed:
First iteration of Generator design - VQVAE works

Azrail09Code committed:
Added information about the PixelCNN

Azrail09Code committed:
Updated model for PixelCNN performance

Azrail09Code committed:
Updating document images for new latent_dims

Azrail09Code committed:
Added better explanation and updated values

Azrail09Code committed:
Merged to computer for model upload

Azrail09Code committed:
Loaded models - generation requires further work

Azrail09Code committed:
Add information on algorithm dependencies

This section was designed to justify the use of numpy for training the pixel_cnn class during fitting.

Azrail09Code committed:
Added new models - final time

Added the final models, however, the generation is unreliable and only sometimes provides generative characteristics.

Azrail09Code committed:
Compiled functions into helper file

Converted file structure to run from internally saved models, and split the main functions into main_functions.py.

SiyuLiu0329 commented:
Difficulty Hard

Functionality
- VQ: OK
- Prior: Quality could be better but OK

Code
- Structure: OK

README 
- Content: OK
- Please include a reference s section at the end and list them all
- It would be better to list dependencies instead of writing a descriptive paragraph
- How was the data normalised?

PR Target OK

General Comments
- Commit messages: OK
- Commit content: Please do not commit checkpoints

shakes76 commented:
## TF/Torch Usage
Adequate use and implementation
## Good Practice (Design/Commenting)
Good spacing and comments
No Header blocks -1
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage 
Module present
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
Good Description and Comments 
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Please remove model files, these will clog up the repository. Your pull request will not be merged until they are removed. Feedback marks will be awarded when these are removed. -2
Request Description OK

Azrail09Code committed:
Removed Model Files

As per Shakes74 Patternflow merging requirements, the pre-trained model files have been removed.

Azrail09Code commented:
Model files have been removed.

Azrail09Code commented:
Hi Shakes,

I’ve removed the mode files now. Thanks for the heads up.

Regards,
Robert

On 22 Nov 2021, at 10:41 am, Shakes ***@***.***> wrote:

﻿

TF/Torch Usage

Adequate use and implementation

Good Practice (Design/Commenting)

Good spacing and comments
No Header blocks -1

Algorithm

Driver Script present
Shows Usage & Demo & Visualisation & Data usage
Module present
No Data leakage
Difficulty: Hard

Commit Log

Meaningful commit messages
Progressive commits used

Documentation

Good Description and Comments
Markdown used PDF submitted

Pull Request

Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch)
Please remove model files, these will clog up the repository. Your pull request will not be merged until they are removed. Feedback marks will be awarded when these are removed. -2
Request Description OK

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/shakes76/PatternFlow/pull/317#issuecomment-974947446>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AKKNZBWZVE3NA6GR3HR6C4TUNGGUJANCNFSM5HDFQNCQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

shakes76 commented:
Feedback marks awarded.
